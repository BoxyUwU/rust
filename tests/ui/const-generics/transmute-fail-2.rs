#![feature(transmute_generic_consts)]
#![feature(generic_const_exprs)]
#![allow(incomplete_features)]

fn bar<const W: bool, const H: usize>(v: [[u32; H]; W]) -> [[u32; W]; H] {
    //~^ ERROR the constant `W` is not of type `usize`
    unsafe {
        std::mem::transmute(v)
    }
}

fn main() {}
