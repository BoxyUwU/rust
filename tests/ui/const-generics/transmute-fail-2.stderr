error: the constant `W` is not of type `usize`
  --> $DIR/transmute-fail-2.rs:5:42
   |
LL | fn bar<const W: bool, const H: usize>(v: [[u32; H]; W]) -> [[u32; W]; H] {
   |                                          ^^^^^^^^^^^^^ expected `usize`, found `bool`

error: aborting due to previous error

